cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -917,13 +924,42 @@ runQueryStakeAddressInfoCmd
 writeStakeAddressInfo
   :: Maybe (File () Out)
   -> DelegationsAndRewards
+  -> Map StakeAddress Lovelace -- ^ deposits
   -> ExceptT QueryCmdError IO ()
-writeStakeAddressInfo mOutFile delegsAndRewards =
-  case mOutFile of
-    Nothing -> liftIO $ LBS.putStrLn (encodePretty delegsAndRewards)
-    Just (File fpath) ->
-      handleIOExceptT (QueryCmdWriteFileError . FileIOError fpath)
-        $ LBS.writeFile fpath (encodePretty delegsAndRewards)
+writeStakeAddressInfo
+  mOutFile
+  (DelegationsAndRewards (stakeAccountBalances, stakePools))
+  stakeDelegDeposits =
+    case mOutFile of
+      Nothing -> liftIO $ LBS.putStrLn (encodePretty jsonInfo)
+      Just (File fpath) ->
+        handleIOExceptT (QueryCmdWriteFileError . FileIOError fpath)
+          $ LBS.writeFile fpath (encodePretty jsonInfo)
```

Comment:
You should be able to use this function here? https://github.com/input-output-hk/cardano-api/blob/newhoggy/new-shelleyToAlonzoEraToShelleyToBabbageEra-function/cardano-api/internal/Cardano/Api/IO.hs/#L115-L123